File: plugin/src/main/java/com/violeth/blockjs/blockjs/binds/BlockBinds.java
package com.violeth.blockjs.blockjs.binds;

import java.util.HashMap;

public class BlockBinds {
    public int x;
    public int y;
    public int z;
    public HashMap<Integer, BlockBind> map = new HashMap<>();
    public BlockBinds(int x, int y, int z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
    public BlockBinds(int x, int y, int z, BlockBind bind) {
        this(x, y, z);
        this.map.put(bind.id, bind);
    }
}

File: plugin/src/main/kotlin/io/github/blockjs/plugin/binds/BlockBinds.kt
package io.github.blockjs.plugin.binds

class BlockBinds(var x: Int, var y: Int, var z: Int) {
    var map: HashMap<Int, BlockBind> = HashMap()

    constructor(x: Int, y: Int, z: Int, bind: BlockBind) : this(x, y, z) {
        map[bind.id] = bind
    }
}
